### ISSUE TITLE
stm32: g0b1re: Log/Shell subsys with serial uart buggy after #38432

### ISSUE THREAD

**Issue by ycsin** [zephyrproject-rtos/zephyr#38519]
**Describe the bug**
The log and shell subsys is acting strangely (sluggish and buggy) after the merge of #38432. Most notably, the shell responds only after second keystrokes. For example if `a` is pressed, nothing appears on the screen. Pressing `a` again will have two `a`s on the shell.

I'm seeing this on my G0B1RE custom board, reverting #38432 solves the issue.

**To Reproduce**
I've tried the `hello_world` with `CONFIG_SHELL` enabled on the `nucleo_g0b1re`, same issue as my custom board

**Environment (please complete the following information):**
 - OS: Win10
 - Toolchain: GNUARMEMB
 - f83dc928f6

**Logs**
```
I: Starting bootloader
I: Primary image: magic=good, swap_type=0x1, copy_done=0x3, image_ok=0x1
I: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: primary slot
I: Swap type: none
I: Bootloader chainload address offset: 0x10000
I: Jumping to the first image slot
***** delaying boot 2000ms (per build configuration) *****





uild time: Sep 15 2021 16:52:30 Malaysia!





1: Turn on modem

32m[00:00:04.011,000] <inf> hw_manager: Starting modem
gmoc:~ $
```

**Comment by ycsin** [comment]
@erwango @FRASTM @gmarull

**Comment by erwango** [comment]
Working on it

**Comment by erwango** [comment]
Fixed by #38686
